src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -640,6 +642,492 @@ public static SyntaxToken XmlTextLiteral(SyntaxTriviaList leading, string text,
+        /// <param name="textTokens">A list of tokens used for the value of the xml text attribute.</param>
+        public static XmlTextAttributeSyntax XmlTextAttribute(string name, params SyntaxToken[] textTokens)
+        {
+            return XmlTextAttribute(XmlName(name), SyntaxKind.DoubleQuoteToken, TokenList(textTokens));
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml text attribute.
+        /// </summary>
+        /// <param name="name">The name of the xml text attribute.</param>
+        /// <param name="quoteKind">The kind of the quote token to be used to quote the value (e.g. " or ').</param>
+        /// <param name="textTokens">A list of tokens used for the value of the xml text attribute.</param>
+        public static XmlTextAttributeSyntax XmlTextAttribute(string name, SyntaxKind quoteKind, SyntaxTokenList textTokens)
+        {
+            return XmlTextAttribute(XmlName(name), SyntaxKind.DoubleQuoteToken, textTokens);
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml text attribute.
+        /// </summary>
+        /// <param name="name">The name of the xml text attribute.</param>
+        /// <param name="quoteKind">The kind of the quote token to be used to quote the value (e.g. " or ').</param>
+        /// <param name="textTokens">A list of tokens used for the value of the xml text attribute.</param>
+        public static XmlTextAttributeSyntax XmlTextAttribute(XmlNameSyntax name, SyntaxKind quoteKind, SyntaxTokenList textTokens)
+        {
+            return XmlTextAttribute(
+                name,
+                Token(quoteKind),
+                textTokens,
+                Token(quoteKind))
+                .WithLeadingTrivia(Whitespace(" "));
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml element that spans multiple text lines.
+        /// </summary>
+        /// <param name="localName">The name of the xml element.</param>
+        /// <param name="content">A list of syntax nodes that represents the content of the xml multi line element.</param>
+        public static XmlElementSyntax XmlMultiLineElement(string localName, SyntaxList<XmlNodeSyntax> content)
+        {
+            return XmlMultiLineElement(XmlName(localName), content);
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml element that spans multiple text lines.
+        /// </summary>
+        /// <param name="name">The name of the xml element.</param>
+        /// <param name="content">A list of syntax nodes that represents the content of the xml multi line element.</param>
+        public static XmlElementSyntax XmlMultiLineElement(XmlNameSyntax name, SyntaxList<XmlNodeSyntax> content)
+        {
+            return XmlElement(
+                XmlElementStartTag(name),
+                content.Insert(0, XmlNewLine()).Add(XmlNewLine()),
```

Comment:
@AnthonyDGreen So I have to take an explicit string in all such methods and their callers within the SyntaxFactory class. But which kind of new line text should be given to the XmlNewLine method when using it in the SyntaxFactory?